File: iOSWebBox/CoconutSDK/Sources/CoconutSDK/Config/Environment.swift
import Foundation

public enum Environment: String {
    case dev
    case test
    case staging
    case prod

    public var displayName: String {
        switch self {
        case .dev: return "Development"
        case .test: return "Testing"
        case .staging: return "Staging"
        case .prod: return "Production"
        }
    }

    public var defaultH5Domain: String {
        switch self {
        case .dev: return "http://192.168.3.49:5174"
        case .test: return "http://192.168.1.100:5174"
        case .staging: return "https://staging-h5.example.com"
        case .prod: return "https://h5.example.com"
        }
    }

    public var defaultApiDomain: String {
        switch self {
        case .dev: return "http://192.168.3.49:8080"
        case .test: return "http://192.168.1.100:8080"
        case .staging: return "https://staging-api.example.com"
        case .prod: return "https://api.example.com"
        }
    }

    public var isDebug: Bool {
        return self == .dev || self == .test
    }
}

File: AndroidWebBox/coconut-sdk/src/main/java/com/sniper/coconut/config/Environment.kt
package com.sniper.coconut.config

/**
 * Environment configuration
 *
 * Defines different runtime environments for the SDK.
 * Each environment has its own H5 domain, API domain, and feature flags.
 */
enum class Environment(
    val displayName: String,
    val defaultH5Domain: String,
    val defaultApiDomain: String
) {
    DEV(
        displayName = "Development",
        // localhost:5174 + `adb reverse tcp:5174 tcp:5174` —— works on any network
        // (emulator + USB real device) without hardcoding a LAN IP. For Wi-Fi real
        // device testing, override at app startup or use STAGING/PROD.
        defaultH5Domain = "http://localhost:5174",
        defaultApiDomain = "http://localhost:8080"
    ),
    TEST(
        displayName = "Testing",
        defaultH5Domain = "http://192.168.1.100:5174",
        defaultApiDomain = "http://192.168.1.100:8080"
    ),
    STAGING(
        displayName = "Staging",
        defaultH5Domain = "https://staging-h5.example.com",
        defaultApiDomain = "https://staging-api.example.com"
    ),
    PROD(
        displayName = "Production",
        defaultH5Domain = "https://h5.example.com",
        defaultApiDomain = "https://api.example.com"
    );

    val isDebug: Boolean get() = this == DEV || this == TEST
}
